### PR TITLE
add remoteLog function for gathering gui logs

### PR DESCRIPTION
**Pull request by PabloBuchu**
### Pull Request Description

Example implementation for `remoteLog` function. It uses JWT `accessToken` obtained by the authorization module to sign requests send to `/logs` endpoint of currently used environment.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: lib/rust/ensogl/pack/js/src/runner/index.ts [hunk 1]
```diff
@@ -197,14 +197,20 @@ export class App {
      * (like wasm compilation) should be aborted. Currently, we initialize application fully
      * before using stop, but in future we need to support interruption feature. */
     stopped = false
+    accessToken: string | null = null
```
Comment (wdanilo) [inline lib/rust/ensogl/pack/js/src/runner/index.ts]: `App.accessToken` does not explain what it is for. Also, I'm not sure if EnsoGL Pack should have a notion of an access token. It is meant to be like a web-pack replacement, which allows shipping any EnsoGL Apps easily. So this is probably not the right place to add such things.

File: lib/rust/ensogl/pack/js/src/runner/index.ts [hunk 2]
```diff
@@ -238,12 +244,40 @@ export class App {
...
+                            `${errorMessage} Failed to read response: ${String(error)}.`
+                        )
+                    }
+                }
+            }
+        } catch (error) {
+            logger.error(error)
+            throw error
+        }
     }
```
Comment (wdanilo) [inline lib/rust/ensogl/pack/js/src/runner/index.ts]: Hmmmm, EnsoGL Pack has a logger, and you can connect any backend to it, including remote logging. For example @mwu-tow connected a file logger to it. I dont think remote logging should be part of EnsoGL Pack.

**Review by wdanilo — CHANGES_REQUESTED** [review]

File: app/ide-desktop/lib/content/src/remoteLog.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+/** URL address where remote logs should be sent. */
+const REMOTE_LOG_URL = new URL(`${authConfig.ACTIVE_CONFIG.apiUrl}/logs`)
+
+/** Helper class facilitating sending logs to a remote. */
+export class RemoteLogger {
+    /** Initialize a new instance.
+     * @param accessToken - JWT token used to authenticate within the cloud. */
+    constructor(public accessToken: string) {
+        this.accessToken = accessToken
+    }
```
Comment (wdanilo) [inline app/ide-desktop/lib/content/src/remoteLog.ts]: sections missing

**Review by wdanilo — APPROVED** [review]

**Comment by xvcgreg** [comment]
@mwu-tow What's the delay here about?

**Comment by mwu-tow** [comment]
Waiting for the cloud team to test. cc @indiv0

**Comment by PabloBuchu** [comment]
I'll QA this one today

**Comment by PabloBuchu** [comment]
@mwu-tow how could we add any project identifier to metadata? `projectName` or `projectUuid`?

**Comment by PabloBuchu** [comment]
Looks good, logs are being pushed properly. I added small improvement for now on backend to associate user email with every entry. Also created issue for improving the log quality with currently opened project https://github.com/enso-org/cloud-v2/issues/537 I think it won't be trivial but we need that also

**Pull request merged**